src/plugins/iconv/iconv.c

```diff
@@ -184,8 +232,15 @@ int elektraIconvGet(Plugin *handle, KeySet *returned, Key *parentKey)
 		ksDel (pluginConfig);
 		return 1;
 	}
-
-	if (!kdbbNeedsUTF8Conversion(handle)) return 0;
```

Comment:
Yes, but your statement is only valid if validation is happening at all.
Iconv is not only for checking, but also for converting.

Honestly, the optimization of from==to is not really important anymore.
If you do not want to use iconv, you would not insert the plugin at all.

But, is it necessary to use iconv the whole text twice for 
validation+transformation? This should be at least documented.